crates/iceberg-catalog/src/service/storage/error.rs

```diff
@@ -16,6 +16,12 @@ pub enum ValidationError {
     },
     #[error(transparent)]
     FileIoError(#[from] FileIoError),
+    #[error("{reason}")]
+    InvalidS3Location {
```

Comment:
InvalidLocation with maybe an internal marker for what storage it belongs to? Else we may end up with a variant per supported backend

Reply:
Done